### ISSUE TITLE
Torrent size as hard filtering criteria

### ISSUE THREAD

**Issue by satcos** [autodl-community/autodl-irssi#201]
When two filters have same filtering criteria but different size limit and action, the filter with lower priority will never be triggered. Lower priority filter will be triggered only after the higher one hitting the count limit.

This also happens if higher priority filter is narrow and lower priority one is wider filter

Example
Filter1: Movies/*, Year- 2019 and 2020, Size 15GB to 30GB
Filter2: Movies/*, Year- 2000 to 2020, Size 5GB to 10GB

A torrent of Movies/*, Year 2020, Size 7.5GB should trigger Filter 2 but this is not happening.

Need to fix this issue as torrent size plays a major roll in decision making.

**Comment by thebigmunch** [comment]
This has come up before. This isn't really going to happen without someone else doing the work to restructure parts of the code.

**Comment by satcos** [comment]
I would like to take up the work, If you could share details of your test environment it will help me get started quickly
